src/v/utils/bottomless_token_bucket.cc

```diff
@@ -66,3 +71,22 @@ void bottomless_token_bucket::refill(const clock::time_point now) noexcept {
         }
     }
 }
+
+// namespace fmt {
+template<>
+typename fmt::basic_format_context<fmt::appender, char>::iterator
+fmt::formatter<bottomless_token_bucket, char, void>::format<
+  fmt::basic_format_context<fmt::appender, char>>(
+  const bottomless_token_bucket& v,
+  fmt::basic_format_context<fmt::appender, char>& ctx) const {
+    return fmt::format_to(
+      ctx.out(),
+      "{{quota:{}, width:{}, tokens:{}, burst:{}, last_check:{}}}",
+      v._quota,
+      v._width.count(),
+      v._tokens,
+      v._burst,
+      v._last_check.time_since_epoch().count());
```

Comment:
When I look at the log with the token bucket contents dumped, looking at that field I can reason about whether it has changed since the last time or not. That was the main reason to get it there. I haven't done anything more complicated like 2. or 3. to keep it simple and avoid paying performance price for unclear reason.

Reply:
> to keep it simple

it's not simple now because last_check has no relation to wall-clock time but we are printing out its value relative to the epoch. what other options are available?

> paying performance price

that would need to be judged on a case by case basis. some solutions are unlikely to introduce anything of concern.

Reply:
> it's not simple now because last_check has no relation to wall-clock time but we are printing out its value relative to the epoch

and that is good enough for the purpose of seeing if it has changed or not, and also comparing relative changes

> what other options are available?

With regard to what is printed, not many. We can't convert it to wall clock and therefore print a human comprehensible timestamp. To use low-resolution system clock with the sole purpose of nice log printout is an absolute overkill. The only option here is to display duration since a certain point in time, like
* the clock epoch (done now)
* another fixed timepoint (eg 2023-01-01)
* a not-that-fixed timepoint like process start time
  * or like the class object creation time
* the last call to the formatter

IMO none of those options help in the task of seeing the change in the time point, they only can lessen the number of digits in the output.

In terms of how it is printed, there is an option is to represent the time since (epoch or whatever) in human friendly time units, like seconds or HH:MM:SS. This does not add value to that information, but makes it easier to reason about. I think this is something I can do.

Reply:
> and that is good enough for the purpose of seeing if it has changed or not, and also comparing relative changes

it's fine to log something just to have visibility in what is going on, so i agree that it is good enough.

what is problematic is the the line of code. you could add a comment to fix this, such as:

```
// computing time_since_epoch here has no meaning, we just care about seeing this value change
```

or, you could do something like this

```
// we only care about how this value is changing since it is a steady clock
std::chrono::duration_cast<std::chrono::milliseconds>(v._last_check - clock::time_point::min()).count();
```

which tbh is probably exactly the same thing, but it expresses what is going on more clearly imo.